faunadb-common/src/main/java/com/faunadb/common/models/tags/Tag.java

```diff
@@ -0,0 +1,84 @@
+package com.faunadb.common.models.tags;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Tag {
+
+    private static final int KEY_CHAR_LIMIT = 40;
+    private static final int VALUE_CHAR_LIMIT = 80;
+
+    private static final Pattern ALLOWABLE_CHARACTER_REGEX = Pattern.compile("^\\w+$");
+    private String key;
+    private String value;
+
+    public Tag() {
+        this.key = "";
+        this.value = "";
+    }
+
+    public Tag(String key, String value) {
+        this.key = getValidatedKey(key);
+        this.value = getValidatedValue(value);
+    }
+
+    @Override
+    public int hashCode() {
+        // For the purposes of hashing, two Tag objects with the same key should collide
+        return Objects.hash(this.key);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        // Perpetually borrowed from https://tinyurl.com/2p99bh7m
```

Comment:
tiny urls change.

Let's use a different link.

This is also a pretty standard .equals implementation. I'm OK omitting this.

Reply:
Default just uses object identity, whereas I explicitly wanted the key/value comparison. Reverted back to the original url, wasn't aware tinyURLs can actually change! That's good to know.

Reply:
I just mean that the way you are overriding equals is a pretty standard way to do it.

I don't think the reference is needed.

Reply:
Ah, I see. Removed.